### PR TITLE
[SHELL32] IShellFolderView default interface stub

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose
Implement CDefView default callback interface.
JIRA issue: [CORE-14976](https://jira.reactos.org/browse/CORE-14976)

- Add _GetDefViewCB method in CDefView.
- Implement CDefViewCB class stub.

### CODE REVIEW

File: dll/win32/shell32/CDefView.cpp [hunk 1]
```diff
@@ -389,6 +390,12 @@ CDefView::~CDefView()
 {
     TRACE(" destroying IShellView(%p)\n", this);
 
+    if (m_pShellFolderViewCB)
```
Comment (learn-more) [inline dll/win32/shell32/CDefView.cpp]: Why is this here?
`m_pShellFolderViewCB` is a smart pointer, so no need to manage it by hand?

File: dll/win32/shell32/CDefView.cpp [hunk 2]
```diff
@@ -3265,11 +3272,16 @@ HRESULT CDefView::_MergeToolbar()
 
 HRESULT CDefView::_DoFolderViewCB(UINT uMsg, WPARAM wParam, LPARAM lParam)
 {
-    if (m_pShellFolderViewCB)
+    if (!m_pShellFolderViewCB)
     {
-        return m_pShellFolderViewCB->MessageSFVCB(uMsg, wParam, lParam);
+        m_pShellFolderViewCB = _GetDefViewCB(this);
+        m_pShellFolderViewCB->AddRef();
```
Comment (learn-more) [inline dll/win32/shell32/CDefView.cpp]: Why is there an extra `AddRef` here?
Assigning to a smart pointer automatically adds a reference.

File: dll/win32/shell32/CDefView.cpp [hunk 3]
```diff
@@ -3326,3 +3338,71 @@ HRESULT WINAPI SHCreateShellFolderView(const SFV_CREATE *pcsfv,
     *ppsv = psv.Detach();
     return hRes;
 }
+
+/* CDefView callback interface */
+class CDefViewCB : public IShellFolderViewCB
```
Comment (learn-more) [inline dll/win32/shell32/CDefView.cpp]: I'd suggest to use ATL, for example `CComObjectRootEx`:
https://git.reactos.org/?p=reactos.git;a=blob;f=dll/shellext/zipfldr/CFolderViewCB.cpp;hb=e8d16d0a7d7c5adc70e40e66db15ed372647345b#l11

**Comment by yagoulas** [comment]
What does this fix? Also, this isn't how callbacks are meant to be used. Callbacks exist so that creators of this object can customize its behaviour. For example an IShellFolder can return an IShellView with slightly customized behaviour.

**Comment by katahiromz** [comment]
Closed.
